homeassistant/components/swiss_public_transport/__init__.py

```diff
@@ -1,38 +1,75 @@
 """The swiss_public_transport component."""
+import logging
+from types import MappingProxyType
+from typing import Any
+
+from opendata_transport import OpendataTransport
+from opendata_transport.exceptions import OpendataTransportError
 import voluptuous as vol
 
 from homeassistant import config_entries, core
 from homeassistant.components.sensor import PLATFORM_SCHEMA
-from homeassistant.const import CONF_NAME
+from homeassistant.const import Platform
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.typing import ConfigType
 
-from .const import CONF_DESTINATION, CONF_START, DEFAULT_NAME, DOMAIN
+from .const import CONF_DESTINATION, CONF_START, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Required(CONF_DESTINATION): cv.string,
         vol.Required(CONF_START): cv.string,
+        vol.Required(CONF_DESTINATION): cv.string,
     }
 )
 
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
 
 async def async_setup_entry(
     hass: core.HomeAssistant, entry: config_entries.ConfigEntry
 ) -> bool:
     """Set up platform from a ConfigEntry."""
     hass.data.setdefault(DOMAIN, {})
-    hass.data[DOMAIN][entry.entry_id] = entry.data
 
-    # Forward the setup to the sensor platform.
-    hass.async_create_task(
-        hass.config_entries.async_forward_entry_setup(entry, "sensor")
-    )
-    return True
+    config = entry.data
 
+    hass.data[DOMAIN][entry.entry_id] = config
+    hass.data[DOMAIN][
+        f"{entry.entry_id}_opendata_client"
+    ] = await swiss_public_transport_opendata_client(hass, config)
 
-async def async_setup(hass: core.HomeAssistant, config: ConfigType) -> bool:
-    """Set up the platform from yaml configuration."""
-    hass.data.setdefault(DOMAIN, {})
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
     return True
+
+
+async def swiss_public_transport_opendata_client(
```

Comment:
You can just put this function in async_setup_entry, it's not that big (and it gives a better overview of the setup)